### ISSUE TITLE
[rllib] Custom SampleCollector can't be set

### ISSUE THREAD

**Issue by jbedorf** [ray-project/ray#17431]
### What is the problem?

There is a naming issue between the default configuration flags and the flag that the sampler uses to determine if the user specifies a custom SampleCollector class. This results in not being able to set the custom class in the training configration. 

* Configuration definition "sample_collector": https://github.com/ray-project/ray/blob/ray-1.5.0/rllib/agents/trainer.py#L287
* Reading the configuration via "sample_collector_class": https://github.com/ray-project/ray/blob/ray-1.5.0/rllib/evaluation/rollout_worker.py#L682  (and line 704)

Ray version: 1.5 / master

### Reproduction (REQUIRED)

The below script should not train, but it does because 'MyCollector' is never initialized. Either "sample_collector" in 'trainer.py" should become "sample_collector_class" or vice versa in rollout_worker.py

```
import ray
from ray import tune

from ray.rllib.evaluation.collectors.sample_collector import SampleCollector


class MyCollector(SampleCollector):
    def __init__(
        self,
        policy_map,
        clip_rewards,
        callbacks,
        multiple_episodes_in_batch,
        rollout_fragment_length,
        count_steps_by,
    ):
        """Initializes a crashing  instance."""
        raise Exception("I am initialized.")


ray.init()
tune.run(
    "PPO",
    stop={"training_iteration": 2},
    config={"env": "CartPole-v0", "num_workers": 1, "sample_collector": MyCollector},
)
```

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by krfricke** [comment]
Thanks for the thorough repro script and pointing to the code! This will be fixed in #17460

**Comment by jbedorf** [comment]
Great, thanks!
